tools/module_dependencies.py

```diff
@@ -0,0 +1,82 @@
+__all__ = ['modules_dependent_on', 'dependency_graph']
+
+
+import numpy as np
+import importlib
+import multiprocessing
+import functools
+import sys
+
+
+package = 'skimage'
+included_mods = [
+    '_shared',
+]
+excluded_mods = [
+    '__version__',
+]
+
+
+@functools.cache
+def _pkg_modules() -> list[str]:
+    pkg = importlib.import_module(package)
+
+    members = {f'{package}.{attr}' for attr in dir(pkg)}
+    included_mods_full = {f'{package}.{mod}' for mod in included_mods}
+    excluded_mods_full = {f'{package}.{mod}' for mod in excluded_mods}
+
+    return (members | set(included_mods_full)) - set(excluded_mods_full)
+
+
+@functools.cache
+def _pkg_modules_index():
+    mods = sorted(_pkg_modules())
+    return {mod: index for index, mod in enumerate(mods)}
+
+
+def _import_dependencies(module: str | list[str]) -> set[str]:
+    importlib.import_module(module)
+    importlib.import_module(f"{module}.tests")
+
+    pkg_modules = _pkg_modules()
+    pkg_sys_modules = {mod for mod in sys.modules if package in mod}
+
+    return pkg_modules & pkg_sys_modules
```

Comment:
Assuming `_pkg_modules()` ends up returning a sorted list, can this just be:
```suggestion
    return set(_pkg_modules()) & set(sys.module)
```
?